src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
+
+namespace Elastic.Apm.Metrics.MetricsProvider
+{
+	public class SystemTotalCpuProvider : IMetricsProvider
+	{
+		private const string SystemCpuTotalPct = "system.cpu.total.norm.pct";
+		private readonly IApmLogger _logger;
+
+		public SystemTotalCpuProvider(IApmLogger logger) => _logger = logger.Scoped(nameof(SystemTotalCpuProvider));
+
+		private bool _firstTotal = true;
+
+		private bool _getTotalProcessorTimeFailedLog;
+		private TimeSpan _lastCurrentTotalProcessCpuTime;
+		private DateTime _lastTotalTick;
+		private Version _processAssemblyVersion;
+		private PerformanceCounter _processorTimePerfCounter;
+
+		public int ConsecutiveNumberOfFailedReads { get; set; }
+		public string NameInLogs => "total system CPU time";
+
+		public IEnumerable<Sample> GetValue()
+		{
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				if (_processorTimePerfCounter == null)
+					_processorTimePerfCounter = new PerformanceCounter("Processor", "% Processor Time", "_Total");
+
+				var val = _processorTimePerfCounter.NextValue();
+
+				return new List<Sample> { new Sample(SystemCpuTotalPct, (double)val / 100) };
+			}
+
+			//The x-plat implementation is ~18x slower than perf. counters on Windows
+			//Therefore this is only a fallback in case of non-Windows OSs
+			var timespan = DateTime.UtcNow;
+			TimeSpan cpuUsage;
+
+			foreach (var proc in Process.GetProcesses())
```

Comment:
Wow! Iterating over all the currently running processes! That is very heavy operation. It probably won't work at all considering that agent is running inside web server process which will usually have very low privileges to minimize attack surface. Isn't there a more reliable and faster mechanism on Linux? [What about `/proc/stat`](https://rosettacode.org/wiki/Linux_CPU_utilization)?

Reply:
> That is very heavy operation

You have any numbers? I really appreciate having such comments only with numbers backing it.

This implementation is based on [this comment](https://github.com/dotnet/standard/issues/327#issuecomment-450919113).

Added a linux specific implementation based on `/proc/stat` - so on Windows we use now PerfCounters, and on Linux we use `/proc/stat`, so this is the fallback for everything else (which is a very small subset of scenarios).

With that, this should be addressed, the last question is: should we keep this x-plat fallback implementation? I tested it on macOS extensively and it seem to deliver the same values as Activity monitor does. Also we print a warning about potential inaccuracy (again, only on non-Windows and non-Linux, probably a tiny portion of our user base). I think this implementation still adds value to customers, so I'd keep it.

We can reiterate on it later, but I'd still keep it.